sourcegen-model/src/main/java/io/micronaut/sourcegen/model/ExpressionDef.java

```diff
@@ -677,12 +688,98 @@ record NewArrayOfSize(TypeDef.Array type, int size) implements ExpressionDef {
     /**
      * The new array expression.
      *
-     * @param type       The type
+     * @param type        The type
      * @param expressions The items expression
      * @author Denis Stepanov
      * @since 1.2
      */
     @Experimental
-    record NewArrayInitialized(TypeDef.Array type, List<ExpressionDef> expressions) implements ExpressionDef {
+    record NewArrayInitialized(TypeDef.Array type,
+                               List<ExpressionDef> expressions) implements ExpressionDef {
+    }
+
+    /**
+     * The get property value expression.
+     *
+     * @param instance        The instance
+     * @param propertyElement The property element
+     * @author Denis Stepanov
+     * @since 1.3
+     */
+    @Experimental
+    record GetPropertyValue(ExpressionDef instance,
+                            PropertyElement propertyElement) implements ExpressionDef {
+
+        @Override
+        public TypeDef type() {
+            return TypeDef.of(propertyElement.getType());
+        }
+    }
+
+    /**
+     * The get class expression.
+     *
+     * @param instance The instance
+     * @author Denis Stepanov
+     * @since 1.3
+     */
+    @Experimental
+    record GetClassValue(ExpressionDef instance) implements ExpressionDef {
```

Comment:
I think `ClassGetter` clould be better.

Reply:
I don't like `Getter` name. It reminds me of properties

Reply:
You are right, getter is strange. Can you name it `GetClass` then? I was confused what is a "class value" when I initially read the name.

Reply:
Added invokeGetClass